client/app/hosting/ftp/hosting-ftp.controller.js

```diff
@@ -1,187 +1,191 @@
-            if (moment(hosting.creation).isAfter(moment().subtract(1, "days"))) {
-                $scope.displayRestoreFtp = false;
-            }
-        });
-    }
+            this.condition = this.Hosting.constructor.getPasswordConditions();
+
+            this.Hosting.getSelected(this.$stateParams.productId)
+                .then((hosting) => {
+                    // backup snapshots are made at least one day after service creation, so hide backup option in the meantime
+                    if (moment(hosting.creation).isAfter(moment().subtract(1, "days"))) {
+                        this.displayRestoreFtp = false;
+                    }
+                    return this.Hosting.getOfferCapabilities(_.camelCase(hosting.offer).toLowerCase());
+                })
+                .then((capabilities) => {
+                    this.displayFtpExplorer = _.get(capabilities, "filesBrowser", false);
+                });
+
+            this.loadTab(10, 0, false);
+            this.startPolling();
+        }
 
-    function startPolling () {
-        HostingUser.getTasks($stateParams.productId, {
-            params: { "function": "user/update" }
-        }).then((tasks) => {
-            HostingUser.pollState($stateParams.productId, {
-                id: tasks[0],
-                successSates: ["done", "cancelled"],
-                namespace: "hosting.web.ftp.user.poll"
-            });
-        });
+        resetSearch () {
+            this.search.value = "";
+            this.goSearch();
+        }
 
-        HostingUser.getTasks($stateParams.productId, {
-            params: { "function": "web/restoreSnapshot" }
-        }).then((tasks) => {
-            if (tasks.length > 0) {
-                $scope.disableRestoreFtp = true;
+        goSearch () {
+            if (!_.isEmpty(this.search.value)) {
+                this.loading.search = true;
             }
-        });
-    }
-
-    $scope.loadFtpInformations = (count, offset) => {
-        $scope.loading.ftp = true;
-        loadTab(count, offset, true);
-    };
 
-    $scope.openFtpExplorer = () => {
-        window.open($scope.ftpInformations.ftpExplorer, "_blank");
-    };
+            this.$scope.$broadcast("paginationServerSide.loadPage", 1);
+        }
 
-    $scope.$on(Hosting.events.tabFtpRefresh, () => {
-        $scope.loading.init = true;
-        $scope.hasResult = false;
-        $scope.$broadcast("paginationServerSide.reload");
-    });
+        loadFtpInformations (count, offset) {
+            this.loading.ftp = true;
+            this.loadTab(count, offset, true);
+        }
 
-    $scope.$on(Hosting.events.tasksChanged, () => {
-        $scope.disableRestoreFtp = true;
-    });
+        loadTab (count, offset, needUsers) {
+            this.Hosting.getTabFTP(this.$stateParams.productId, count, offset, needUsers, this.search.value)
+                .then((ftpInformations) => {
+                    if (!_.isEmpty(ftpInformations.list.results)) {
```

Comment:
I would have said`if (ftpInformations  != null)` then we know that it has `list.results`